source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -12,6 +15,16 @@ UdpProxyFilter::UdpProxyFilter(Network::UdpReadFilterCallbacks& callbacks,
     : UdpListenerReadFilter(callbacks), config_(config),
       cluster_update_callbacks_(
           config->clusterManager().addThreadLocalClusterUpdateCallbacks(*this)) {
+  bool check_v4only =
+      callbacks.udpListener().localAddress()->ip()->version() == Network::Address::IpVersion::v4;
+  if (config_->usingOriginalSrcIp() &&
+      !Api::OsSysCallsSingleton::get().supportsIpTransparent(check_v4only)) {
+    throw Network::CreateListenerException(
```

Comment:
I think this should actually be checked inside the the config creation for the UDP proxy, not here on filter creation. This way it will be checked in config load. In that case you can throw an EnvoyException and it will be a normal error. You should also add tests for this case.

Reply:
Ok. I have a question. Originally, I would like to check it on config creation for the UDP proxy. But I could not find how to know the user set the listener's IP as IPv4 or IPv6 in this config creation. That is why I change the checking from config creation to filter creation. Could you give your advise about that? Should I add one more option for that? Or? :)

Reply:
Sorry I think I missed that. Why do we have to check that? Just because we don't know whether v4 or v6 is working until we check it? Can we just check both? If we need to actually check this at runtime I don't think we can actually raise an exception here. I think we need to just drop the datagram and have a stat or something.

Reply:
Actually, it is added due to following @danzh2010 's comment : https://github.com/envoyproxy/envoy/pull/12586#discussion_r473292085

So, I check more linux kernel's implementation, and the ipv6 feature can be disabled during kernel compilation and can be a kernel module so it can be loaded after booting and can be disabled by grub' configuration even if the linux kernel is compiled with ipv6 feature.

I think that there is a advantage that disable the ipv6 on modern common linux distribution.
That is only memory usage but it is not much issue because the memory usage of ipv6 is not so much.

What do you think about that? Should I consider ipv6 disabled environment?

Reply:
Or how about revert this change and mention the ipv6 stack should be enabled on linux kernel at this option's manual?

Reply:
I don't have a strong opinion on this one way or the other, but we just need to make sure the error handling works correctly. If you check this and it can fail at datagram receive time, you need to drop the packet and stat. If you check at config time you can fail then. Up to you as long as we clearly document it.

Reply:
Ok. Thanks your opinion. :) I will revert previous change and I will mention on document that the ipv6 stack is a mandatory for this new option.